mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1627,7 +1631,7 @@ impl TakerSwap {
                 );
                 let swpmsg_watcher = SwapWatcherMsg::TakerSwapWatcherMsg(watcher_data);
                 let htlc_keypair = self.taker_coin.derive_htlc_key_pair(&self.unique_swap_data());
-                watcher_broadcast_abort_handle = Some(broadcast_swap_message_every(
+                watcher_broadcast_abort_handle = Some(broadcast_swap_message_every_with_initial_delay(
```

Comment:
It would be nice to limit names length with 25 or 30 symbols
Too long name encouraged me to suggest something like that ) - if that is possible

```rust
fn repeat_until_aborted<F: Fn() + Send + 'static>(interval_sec: f64, f: F) -> AbortOnDropHandle {
    let fut = async move {
        loop {
            Timer::sleep(interval_sec).await;
            f();
        }
    };
    spawn_abortable(fut)
}
...
watcher_broadcast_abort_handle = Some(repeat_until_aborted(|| broadcast_swap_message(...)))
```

Reply:
Yes it was definitely way too long, I shortened the names of `broadcast_swap_message_every` and `broadcast_swap_message_every_with_initial_delay` to `broadcast_swap_msg_every` and `broadcast_swap_msg_every_delayed`. I kept the rest as it is because then I would need two functions `repeat_until_aborted` and `repeat_until_aborted_delayed` and the second one would be almost as long as what I used, and it would be even more crowded with the `broadcast_swap_message` as a parameter.